### PR TITLE
fix(ui): Fix no access on Project Details

### PR DESCRIPTION
**Pull request by matejminar**
This could happen when the organization had their Open Membership setting disabled and Members visited the project details (either directly via URL or via Organization Stats) to which their team did not have access.

Fixes JAVASCRIPT-247Q

## Before
image

## After
image

### CODE REVIEW

**Comment by priscilawebdev** [comment]
do we really need a button? maybe it would be better if we could update the user's team only by selecting one option 🤷‍♀️ but in case we prefer to keep the button, should we name it "save"? It can be confusing to have two "select a team"

File: static/app/views/projectDetail/projectDetail.tsx [hunk 1]
```diff
@@ -182,6 +201,10 @@ class ProjectDetail extends AsyncView<Props, State> {
       return this.renderProjectNotFound();
     }
 
+    if (!loadingProjects && project && !project?.hasAccess) {
```
Comment (priscilawebdev) [inline static/app/views/projectDetail/projectDetail.tsx:204]:
```suggestion
    if (!loadingProjects && project && !project.hasAccess) {
```

**Comment by matejminar** [comment]
@priscilawebdev I am fixing the permission-related access issue by reusing component that's used in multiple places over the app. If we decide that we want to address the UX of that component, I'd much rather do that in a separate PR. 🙂

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Comment by matejminar** [comment]
@priscilawebdev fwiw, the button changes based on your project selection:
https://user-images.githubusercontent.com/9060071/117289689-2f1c8a80-ae6d-11eb-93bd-92190a5f40c9.mp4
https://github.com/getsentry/sentry/blob/9707fd3f98fa51d5d363cbe34b59aa51588db246/static/app/components/projects/missingProjectMembership.tsx#L78-L111

**Pull request merged**